### PR TITLE
Add details view for analytics

### PR DESCRIPTION
**Pull request by kasparia**

### CODE REVIEW

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js [hunk 1]
```diff
@@ -35,19 +37,52 @@ Oskari.clazz.define('Oskari.framework.bundle.admin-layeranalytics.Flyout',
...
+                toScaleURL += value; // add layer id
+                toScaleURL += '+100'; // add layer opacity
+                toScaleURL += '+'; // add layer default style as empty string
+                if (index !== (stack.layers.length - 1)) {
+                    toScaleURL += ','; // add layer separator if not last layer in stack
+                }
+            }
+
+            return toScaleURL;
+        },
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js:55]: Is there a reason we send this as reference to the component instead of it being part of the component? It only uses the first entry on stack which is problematic so if we want to keep it here we should just send the stack not the whole details. 

Also:
```suggestion
        generateToScaleURL (details) {
            const stack = details.stack[0];
            let toScaleURL = '/?coord=' + stack.x + '_' + stack.y;
            // [layer id]+[opacity as 100]+[style as empty string]
            return toScaleURL + '&mapLayers=' + stack.layers.map(id => id + `+100+`).join(',');
        },
```

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js [hunk 2]
```diff
@@ -35,19 +37,52 @@ Oskari.clazz.define('Oskari.framework.bundle.admin-layeranalytics.Flyout',
...
         updateListing () {
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js:56]: Perhaps rename updateListing() to updateUI() or similar since it's no longer just the listing?

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    return date.toLocaleDateString(dateLocale, localeDateOptions);
+};
+
+const formatTime = (date) => {
+    if (typeof date === 'undefined') {
+        return '--:--:--';
+    }
+    return date.toLocaleTimeString(dateLocale).replace(/\./g, ':');
+};
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:35]: Copy pasted from admin-layereditor? Add a comment so we can maybe move these to a helper that can be shared or atleast remember to update the formatter when needed.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        },
+        {
+            title: '',
+            key: 'action',
+            render: (text, entry) => (
+              <Space size="middle">
+                <a href={ toScaleCallback(entry) } target='_blank'><Message messageKey='flyout.moveToScale' /></a>
+              </Space>
+            )
+        }
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:64]: The third column should render a list of the stack (now just uses the first one).

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+};
+
+const formatTime = (date) => {
+    if (typeof date === 'undefined') {
+        return '--:--:--';
+    }
+    return date.toLocaleTimeString(dateLocale).replace(/\./g, ':');
+};
+
+export const LayerAnalyticsDetails = ({ layerData, closeDetailsCallback, toScaleCallback }) => {
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:37]: Move toScaleCallback to this file. I don't see any reason why it should be on the other file since this is the only one using it.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/instance.js [hunk 6]
```diff
@@ -219,6 +217,10 @@ Oskari.clazz.define('Oskari.framework.bundle.admin-layeranalytics.AdminLayerAnal
             return this.analyticsData;
         },
 
+        getSingleLayerData (id) {
+            return this.analyticsData.find((data) => data.id === id);
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/instance.js:221]: This needs another XHR to work properly. See: https://github.com/nls-oskari/kartta.paikkatietoikkuna.fi/pull/156

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js (hunk 2 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js:56]: Renamed.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js (hunk 1 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js:55]: Moved.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx (hunk 5 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:37]: Moved.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js [hunk 7]
```diff
@@ -35,19 +37,52 @@ Oskari.clazz.define('Oskari.framework.bundle.admin-layeranalytics.Flyout',
 
             this.updateListing();
         },
+        generateToScaleURL (details) {
+            const stack = details.stack[0];
+            let toScaleURL = '/?coord=' + stack.x + '_' + stack.y;
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/Flyout.js:42]: Add `"&zoomLevel=" + stack.z` to get the same zoom level as where the problem occured.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx (hunk 3 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:38]: Added comment here.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/instance.js (hunk 6 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/instance.js]: Check.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+const formatTime = (date) => {
+    if (typeof date === 'undefined') {
+        return '--:--:--';
+    }
+    return date.toLocaleTimeString(dateLocale).replace(/\./g, ':');
+};
+
+const generateToScaleURL = (stack) => {
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:40]: generateLink() would be a better name for this. Remember to add stack.z as zoomLevel param.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+            sortDirections: ['descend', 'ascend', 'descend'],
+            sorter: (a, b) => a.time - b.time,
+            render: (text) => <Space>{ formatTimestamp(text) }</Space>
+        },
+        {
+            title: '',
+            key: 'action',
+            render: (text, entry) => (
+              <Space size="middle">
+                <a href={ generateToScaleURL(entry.stack[0]) } target='_blank'><Message messageKey='flyout.moveToScale' /></a>
```
Comment (ZakarFin) [inline bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:80]: Stack should be looped here, but I'm ok if it's done on another PR.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx (hunk 8 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:40]: Changed.

File: bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx (hunk 9 above)
Comment (kasparia) [reply bundles/paikkatietoikkuna/admin/admin-layeranalytics/LayerAnalyticsDetails.jsx:81]: I'll move it to another PR. I need additional info about the layout of list.

**Comment by ZakarFin** [comment]
Some requested changes are upcoming in the next pr
